tests/components/google_tasks/test_todo.py

```diff
@@ -82,14 +95,87 @@ def mock_api_responses() -> list[dict | list]:
     return []
 
 
+def create_response_object(api_response: dict | list) -> tuple[Response, bytes]:
+    """Create an http response."""
+    return (
+        Response({"Content-Type": "application/json"}),
+        json.dumps(api_response).encode(),
+    )
+
+
+def create_batch_response_object(
+    contentids: list[str], api_responses: list[dict | list | Response]
+) -> tuple[Response, bytes]:
+    """Create a batch response in the multipart/mixed format."""
+    assert len(api_responses) == len(contentids)
+    content = []
+    for api_response in api_responses:
+        status = 200
+        body = ""
+        if isinstance(api_response, Response):
+            status = api_response.status
+        else:
+            body = json.dumps(api_response)
+        content.extend(
+            [
+                f"--{BOUNDARY}",
+                "Content-Type: application/http",
+                f"{CONTENT_ID}: {contentids.pop()}",
+                "",
+                f"HTTP/1.1 {status} OK",
+                "Content-Type: application/json; charset=UTF-8",
+                "",
+                body,
+            ]
+        )
+    content.append(f"--{BOUNDARY}--")
+    body = ("\r\n".join(content)).encode()
+    return (
+        Response(
+            {
+                "Content-Type": f"multipart/mixed; boundary={BOUNDARY}",
+                "Content-ID": "1",
+            }
+        ),
+        body,
+    )
+
+
+def create_batch_response_handler(
+    api_responses: list[dict | list | Response],
+) -> Callable[[Any], tuple[Response, bytes]]:
+    """Create a fake http2lib response handler that supports generating batch responses.
+
+    Multi-part response objects are dynamically generated since they
+    need to match the Content-ID of the incoming request.
+    """
+
+    def _handler(url, method, **kwargs) -> tuple[Response, bytes]:
+        next_api_response = api_responses.pop(0)
+        if method == "POST" and (body := kwargs.get("body")):
+            contentids = [
+                line[len(CONTENT_ID) + 2 :]
+                for line in body.splitlines()
+                if line.startswith(f"{CONTENT_ID}:")
+            ]
+            if contentids:
+                return create_batch_response_object(contentids, next_api_response)
```

Comment:
```suggestion
            content_ids = [
                line[len(CONTENT_ID) + 2 :]
                for line in body.splitlines()
                if line.startswith(f"{CONTENT_ID}:")
            ]
            if content_ids:
                return create_batch_response_object(content_ids, next_api_response)
```

Reply:
Manually applied after merge conflict (resolved in the wrong order)